Redis.cpp

```diff
@@ -159,6 +159,91 @@ bool Redis::tsadd(const char *key, long timestamp, const int value)
   }
 }
 
+int Redis::xack(const char *key, const char *group, const char *id)
+{
+  TRCMD(int, "XACK", key, group, id);
+}
+
+String Redis::xadd(const char *key, const char *id, const char *field,
+                   const char *value)
+{
+  TRCMD(String, "XADD", key, id, field, value);
+}
+
+int Redis::xdel(const char *key, const char *id)
+{
+  TRCMD(int, "XDEL", key, id);
+}
+
+bool Redis::xgroup_create(const char *key, const char *group, const char *id,
+                          bool mkstream)
+{
+  if(mkstream)
+  {
+    TRCMD_EXPECTOK("XGROUP", "CREATE", key, group, id, "MKSTREAM");
+  }
+  else
+  {
+    TRCMD_EXPECTOK("XGROUP", "CREATE", key, group, id);
+  }
+}
+
+int Redis::xgroup_createconsumer(const char *key, const char *group,
+                                 const char *consumer)
+{
+  TRCMD(int, "XGROUP", "CREATECONSUMER", key, group, consumer);
+}
+
+int Redis::xgroup_delconsumer(const char *key, const char *group,
+                              const char *consumer)
+{
+  TRCMD(int, "XGROUP", "DELCONSUMER", key, group, consumer);
+}
+
+int Redis::xgroup_destroy(const char *key, const char *group)
+{
+  TRCMD(int, "XGROUP", "DESTROY", key, group);
+}
+
+bool Redis::xgroup_setid(const char* key, const char *group, const char *id)
+{
+  TRCMD_EXPECTOK("XGROUP", "SETID", key, group, id);
+}
+
+int Redis::xlen(const char *key)
+{
+  TRCMD(int, "XLEN", key);
+}
+
+int Redis::xtrim(const char *key, const char *strategy, const char *exact,
+                 const int threshold, const int count)
+{
+  if(exact == "=" || exact == "~")
```

Comment:
Given that the values for `exact` are a (small) bounded set, I think I'd prefer to see them expressed as an `enum`.